### PR TITLE
Feature: infrastructure

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

This is purely cherry-pick stack from #56 extracting infrastructure code from the rest of noise.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_parameterized [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!groovy
+@Library("Infrastructure") _
+
+properties([
+		parameters([
+			string(name: 'PRODUCT_NAME', defaultValue: 'platform', description: ''),
+			string(name: 'APP', defaultValue: 'job-scheduler', description: ''),
+			string(name: 'TYPE', defaultValue: 'java', description: ''),
+			string(name: 'ENVIRONMENT', defaultValue: 'sandbox', description: 'Environment where code should be build and deployed'),
```
Comment (rkondratowicz) [inline Jenkinsfile_parameterized]: built

File: Jenkinsfile_parameterized [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+			choice(name: 'SUBSCRIPTION', choices: 'sandbox\nnonprod\nprod', description: 'Azure subscriptions available to build in')
```
Comment (rkondratowicz) [inline Jenkinsfile_parameterized]: probably a typo in `nonprod`

Comment (doncem) [reply Jenkinsfile_parameterized]: that's a new line + nonprod. makes `\nnonprod`

Comment (rkondratowicz) [reply Jenkinsfile_parameterized]: :D my bad. Out of interest, can it not be comma separated?

Comment (doncem) [reply Jenkinsfile_parameterized]: looked and thought all this infra code is somewhere in our repos. but did not find. found same references e.g. [here](https://github.com/hmcts/moj-infra-pipeline/blob/8fc8b95cd3d024d40e05c982e4ab551211f24303/Jenkinsfile#L7)

**Review by jasonpaige — APPROVED** [review]

File: infrastructure/main.tf [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+provider "vault" {
+  address = "https://vault.reform.hmcts.net:6200"
+}
+
+locals {
+  vault_section = "${var.env == "prod" ? "prod" : "test"}"
+}
+
+data "vault_generic_secret" "s2s_secret" {
+  path = "secret/${local.vault_section}/ccidam/service-auth-provider/api/microservice-keys/platformJobScheduler"
```
Comment (doncem) [inline infrastructure/main.tf:12]: @rkondratowicz raised a question whether it's a correct usage and should it not be e.g. `platform-job-scheduler` instead. @jasonpaige ?

Comment (jasonpaige) [reply infrastructure/main.tf:12]: iirc that was the same naming convention of what was already there in vault for the other apps... but yeh kind of annoying I must admit. We should correct it when we migrate to Azure Key Vault I think!

**Pull request merged**